### PR TITLE
Update sidecar getting started guide

### PR DESCRIPTION
**Pull request by craigbox**
Bring quality-of-life updates from the ambient getting started guide into the sidecar guide.

Move Gateway API docs to default, and move legacy API docs to /additional-setup/.

Create a page describing the release bundle so it doesn't have to be talked about in detail to people who are touching Istio for the first time on the happy path.

Use Bookinfo from GitHub and not the local bundle as the latest 1.22 build doesn't include the new look and feel.

- [x] Docs
- [x] User Experience

### CODE REVIEW

File: content/en/docs/setup/additional-setup/getting-started-istio-apis/test.sh [hunk 1]
```diff
@@ -73,13 +78,11 @@ _verify_contains get_bookinfo_productpage "<title>Simple Bookstore App</title>"
 # verify Kiali deployment
 _verify_contains snip_view_the_dashboard_1 'deployment "kiali" successfully rolled out'
 
-# Verify Kiali dashboard
+# Verify Kiala dashboard
```
Comment (kfaseela) [inline content/en/docs/setup/additional-setup/getting-started-istio-apis/test.sh:81]:
```suggestion
# Verify Kiali dashboard
```

File: content/en/docs/setup/getting-started/index.md [hunk 2]
```diff
@@ -23,42 +20,27 @@ advanced [deployment models](/docs/ops/deployment/deployment-models/), refer to
 [which Istio installation method should I use?](/about/faq/#install-method-selection)
 FAQ page.
 
-These steps require you to have a {{< gloss >}}cluster{{< /gloss >}} running a
-[supported version](/docs/releases/supported-releases#support-status-of-istio-releases) of Kubernetes ({{< supported_kubernetes_versions >}}). You can use any supported platform, for
-example [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) or
-others specified by the
-[platform-specific setup instructions](/docs/setup/platform-setup/).
+You'll need a Kubernetes cluster to proceed. If you don't have a cluster, you can use [kind](/docs/setup/platform-setup/kind) or any other [supported Kubernetes platform](/docs/setup/platform-setup).
```
Comment (kfaseela) [inline content/en/docs/setup/getting-started/index.md:23]:
```suggestion
You will need a Kubernetes cluster to proceed. If you don't have a cluster, you can use [kind](/docs/setup/platform-setup/kind) or any other [supported Kubernetes platform](/docs/setup/platform-setup).
```

File: content/en/docs/setup/getting-started/index.md [hunk 3]
```diff
@@ -80,24 +62,25 @@ Follow these steps to get started with Istio:
 
 ## Install Istio {#install}
 
-1.  For this installation, we use the `demo`
-    [configuration profile](/docs/setup/additional-setup/config-profiles/). It's
-    selected to have a good set of defaults for testing, but there are other
-    profiles for production or performance testing.
+For this guide, we use the `demo`
+[configuration profile](/docs/setup/additional-setup/config-profiles/). It's
```
Comment (kfaseela) [inline content/en/docs/setup/getting-started/index.md:66]:
```suggestion
[configuration profile](/docs/setup/additional-setup/config-profiles/). It is
```

**Comment by craigbox** [comment]
/retest

I love flakes

**Review by kfaseela — APPROVED** [review]

**Comment by craigbox** [comment]
Is it OK to commit this, @dhawton ?
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/15436/gencheck_istio.io/1817850947750072320/artifacts/check-clean-repo-diff.patch

**Comment by dhawton** [comment]
> Is it OK to commit this, @dhawton ? https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/15436/gencheck_istio.io/1817850947750072320/artifacts/check-clean-repo-diff.patch

Yes.. master should be on release-1.23 until 1.23.0 is cut, we branch 1.23 into its own release branch in this repo, and revert master back to istio's master branch.

**Comment by craigbox** [comment]
And at that point all the snips revert to having master in the path?

**Comment by dhawton** [comment]
> And at that point all the snips revert to having master in the path?

Yes. snips get it from https://github.com/istio/istio.io/pull/15443/files#diff-362b38ffc946b8c2ff34ff45ab8b4292aa26dc06ce1833e92c6c1977af2cc299R28 of data/args.yaml. When this is set back to master and a `make gen` is run, they'll get updated.

**Pull request merged**